extensions/src/ACRE2Arma/common/wrp/landscape.cpp

```diff
@@ -30,8 +30,11 @@ acre::wrp::landscape::~landscape() {
 
 bool acre::wrp::landscape::_read_binary_tree_block(std::istream &stream_, const uint32_t bit_length, const uint32_t block_size, const uint32_t cell_size, const uint32_t block_offset_x, const uint32_t block_offset_y)
 {
-    uint32_t blockSize = static_cast<uint32_t>(ceil((log10(block_size) / log10(2.0)) / 2.0));
-    blockSize = static_cast<uint32_t>(pow(2, 2.0 * blockSize));
+    uint32_t blockSize = 1;
+    if (block_size != 0) {
+        blockSize = static_cast<uint32_t>(ceil((log10(block_size) / log10(2.0)) / 2.0));
+        blockSize = static_cast<uint32_t>(pow(2, 2.0 * blockSize));
+    }
```

Comment:
```suggestion
   // blockSize must remain an integer. If switched to uint32_t, it  can go to -0 with the ceil function and from here to max uint32_t.
   auto blockSize = static_cast<int32_t>(ceil((log10(block_size) / log10(2.0)) / 2.0));
   blockSize = static_cast<int32_t>(pow(2, 2.0 * blockSize));
```

Reply:
log(0) is junk and makes no sense, 
I think it's better to handle this edge case with a condition before it happens instead of trying to handle it after

Reply:
you are right.